src/main/java/com/programmers/java/Main.java

```diff
@@ -0,0 +1,61 @@
+package com.programmers.java;
+
+import com.programmers.java.calculator.calculate.Calculator;
+import com.programmers.java.calculator.calculate.PostfixConverter;
+import com.programmers.java.record.CalculationRecord;
+import com.programmers.java.util.ExpressionTokenizer;
+import com.programmers.java.view.Menu;
+import com.programmers.java.view.Input;
+import com.programmers.java.view.Output;
+
+import java.io.IOException;
+
+public class Main {
+    static Output output = new Output();
+    static Input input = new Input();
+    static Calculator calculator = new Calculator();
+    static CalculationRecord calculationRecord = new CalculationRecord();
+    static ExpressionTokenizer expressionTokenizer = new ExpressionTokenizer();
+    static PostfixConverter postfixConverter = new PostfixConverter();
+
+    public static void main(String[] args) throws IOException {
+        while (true) {
+            output.viewMenu();
+
+            input.enterMenu().ifPresentOrElse(
+                    Main::runCalculator,
+                    () -> {
+                        output.viewEndMessage();
+                        System.exit(0);
+                    }
+            );
+        }
+    }
```

Comment:
System 사용 제한과, while 의 무한 루프 대신 이런식으로 변경 해보면 어떨까 싶어서 코드 제안 드려보았어요.

```suggestion
    public static void main(String[] args) throws IOException {
    	boolean isRunning = true
    	
        while (isRunning) {
            output.viewMenu();

            input.enterMenu().ifPresentOrElse(
                    Main::runCalculator,
                    () -> {
                        isRunning = false;
                        output.viewEndMessage();
                    }
            );
        }
    }
```

Reply:
더 알기 쉬운 코드를 작성하도록 노력할게요 ✏️

Reply:
[65e504c](https://github.com/prgrms-be-devcourse/java-calculator/pull/123/commits/65e504c612e5f6b5a0ce21c9bdf3171032fc5a3d)에 반영
- 람다 바디에 지역변수를 참조하게 되면 final이거나 final처럼 작동해야 하는데 while문 안에서 값을 바꿔주기 때문에 final로 설정할 수 없었어요
- 또 이를 인스턴스 변수로 선언하려고 했는데 메인메서드가 static이기 때문에 사용할 수 없었어요
- 그래서 static으로 바꿨는데 static으로 선언한 경우 멀티 스레드 환경에서 동시 접근 등의 문제가 발생한다고 하여 AtomicBoolean을 사용하였어요

[89ebac2](https://github.com/prgrms-be-devcourse/java-calculator/pull/123/commits/89ebac2360136507c264994300289f6a5d01e234)에 반영
- 계산기는 멀티 스레드 환경이 아니기 때문에 다시 static으로 바꾸게 되었어요 
- 혹시 이 생각의 흐름이 맞을까요?